tests/unit/ordered-set-test.js

```diff
@@ -5,18 +5,220 @@ import { isEmpty } from '@ember/utils';
+    assert.notOk(set.has(foo));
+  });
+
+  test('delete() returns whether object was part of the set', function(assert) {
+    let set = OrderedSet.create();
+
+    set.add('foo');
+
+    assert.strictEqual(set.delete('foo'), true);
+    assert.strictEqual(set.delete('bar'), false);
+  });
+
+  test('isEmpty() returns whether the set has any entries', function(assert) {
+    let set = OrderedSet.create();
+    assert.strictEqual(set.isEmpty(), true, 'Empty ordered set is empty');
+
+    set.add('foo');
+    assert.strictEqual(set.isEmpty(), false, 'Ordered set is not empty');
+  });
+
+  test('has() returns whether an object is part of the set', function(assert) {
+    let set = OrderedSet.create();
+
+    set.add('foo');
+
+    assert.strictEqual(set.has('foo'), true);
+    assert.strictEqual(set.has('bar'), false);
+  });
+
+  test('has() supports non-string objects', function(assert) {
+    let set = OrderedSet.create();
+
+    let foo = { bar: 'baz' };
+    let bar = { baz: 'foo' };
+
+    set.add(foo);
+
+    assert.strictEqual(set.has(foo), true);
+    assert.strictEqual(set.has(bar), false);
+  });
+
+  test('forEach() iterates over all entries', function(assert) {
+    let set = OrderedSet.create();
+
+    set.add('foo');
+    set.add('bar');
+    set.add({ baz: 'qux' });
+
+    let entries = [];
+    set.forEach(entry => entries.push(entry));
+
+    assert.deepEqual(entries, ['foo', 'bar', { baz: 'qux' }]);
+  });
+
+  test('forEach() is called with no context by default', function(assert) {
+    assert.expect(2);
+
+    let set = OrderedSet.create();
+
+    set.add('foo');
+
+    set.forEach(function(entry) {
+      assert.strictEqual(entry, 'foo');
+      assert.strictEqual(this, undefined);
+    });
+  });
+
+  test('forEach() context can be set as second argument', function(assert) {
+    assert.expect(2);
+
+    let set = OrderedSet.create();
+
+    set.add('foo');
+
+    let context = { bar: 'baz' };
+
+    set.forEach(function(entry) {
+      assert.strictEqual(entry, 'foo');
+      assert.strictEqual(this, context);
+    }, context);
+  });
+
+  test('toArray() returns an array of all entries', function(assert) {
+    let set = OrderedSet.create();
+
+    set.add('foo');
+    set.add('bar');
+    set.add({ baz: 'qux' });
+
+    let entries = set.toArray();
+
+    assert.deepEqual(entries, ['foo', 'bar', { baz: 'qux' }]);
+  });
+
+  test('copy() copies all entries into a new set', function(assert) {
+    let set = OrderedSet.create();
+
+    let baz = { baz: 'qux' };
+
+    set.add('foo');
+    set.add('bar');
+    set.add(baz);
+
+    assert.ok(set.has('foo'));
+    assert.ok(set.has(baz));
+
+    let copiedSet = set.copy();
+
+    assert.ok(copiedSet.has('foo'));
+    assert.ok(copiedSet.has(baz));
   });
 
-  test('isEmpty()', function(assert) {
-    let orderedSet = new OrderedSet();
-    assert.equal(true, isEmpty(orderedSet), 'Empty ordered set is empty');
+  test('is compatible with Ember.isEmpty()', function(assert) {
```

Comment:
do we have a better way to reference `isEmpty` than this? (because of new modules)

Reply:
it's only called `Ember.isEmpty()` here to disambiguate from the `isEmpty()` method on the class itself. the testing code is already using the new module imports.

Reply:
You mean in prose?  I don't think we do actually...